static/js/narrow.js

```diff
@@ -187,6 +187,11 @@ exports.activate = function (raw_operators, opts) {
 
             var then_scroll = !preserve_pre_narrowing_screen_position;
 
+            // Select the first unread if it's above the current message ID
+            if (then_select_id > message_list.narrowed.first_unread_message_id()) {
+                then_select_id = message_list.narrowed.first_unread_message_id();
+            }
+
```

Comment:
@showell I think this is a bug worth fixing while we're doing our narrowing project (see the issue for a clear explanation of it).

The logic @AasthaGupta has here should work in the case that we have all the necessary data in the browser for the unread messages; but I think we'll basically want to redo this with your new exact/etc. framework...